### PR TITLE
add slowg analyzer to check inappropriate use of log/slog

### PR DESCRIPTION
**Pull request by rolinh**

### CODE REVIEW

**Review by nathanjsweet — CHANGES_REQUESTED** [review]
Will this analyzer run by default? If so, I'm worried that it will break the cilium repo.
Everything else looks good.

**Comment by rolinh** [comment]
> Will this analyzer run by default?

I've enabled it by default, yes.

> If so, I'm worried that it will break the cilium repo.

In which way? I don't understand your concern as Cilium is not even using log/slog yet 🤔

**Comment by rolinh** [comment]
Converted back to draft, the log functions were renamed from `XXXCtx` to `XXXContext` and thus the linter needs an update.

**Comment by rolinh** [comment]
CI is now failing as there is no tagged release of Go 1.21 that includes [this change](https://github.com/golang/go/commit/3f8b04bfb541127704ec5c33a52294fe3c6ebea4) yet. Unfortunately, it looks like we cannot specify `tip` or a commit sha as the Go version to use in the matrix of tests.

**Comment by rolinh** [comment]
I'll mark the PR as ready to merge. CI should turn green once we update Go 1.21 to the RC.3.

**Review by kaworu — APPROVED** [review]

File: ioreadall/io_readall.go [hunk 1]
```diff
@@ -22,8 +22,9 @@ var ioReadAllPkgs = []string{"io", "ioutil"}
 // Analyzer implements an analysis function that checks for the use of
 // io.ReadAll.
 var Analyzer = &analysis.Analyzer{
-	Name:     "readall",
+	Name:     "ioreadall",
 	Doc:      `check for "io.ReadAll" instances`,
+	URL:      "https://github.com/cilium/linters/ioreadall",
```
[the remark this answers is not in the document]
Comment (rolinh) [reply ioreadall/io_readall.go]: Ah good point. Well, there's no readme in that directory, let's simply use `https://github.com/cilium/linters`.

**Comment by rolinh** [comment]
Go v1.21.0-rc.3 is out so I updated the Go matrix in the CI tests and, as expected, the tests now pass.

@nathanjsweet Could you please check this PR again or give a rationale as to why you think we shouldn't merge it as is?

**Pull request merged**